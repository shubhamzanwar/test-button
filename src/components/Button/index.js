import React, { Component,useState } from 'react'
import * as styles from './index.css';


const Button=({text,buttonType,testId,click})=>{
   
    const [count,setCount]=useState(0);
    const countHandler=()=>{
        setCount(count+1);
        click();
    }
    return (
        <button className={buttonType === 'rect' ? styles.rect : styles.round} data-testid={testId} onClick={countHandler}>
          {text} clicked {count} times.
        </button> 
      )

}
export default Button;

