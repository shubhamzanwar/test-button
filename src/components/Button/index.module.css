.round{
    border-radius: 50%;
}

.rect{
    border: solid 1rem;
}