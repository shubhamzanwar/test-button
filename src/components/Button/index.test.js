import React from 'react';
import Button from './index';
import { render, fireEvent } from '@testing-library/react';

describe('the button component', () => {
    xit('should render correctly', () => {
        const { asFragment } = render(<Button testID="test-btn" type="rect"/>);

        expect(asFragment()).toMatchSnapshot();
    });

   xit('should call the onClick prop when the button is clicked', () => {
        const onClickMock = jest.fn();
        const { getByTestId } = render(<Button testID="test-btn" type="rect" onClick={onClickMock}/>);

        fireEvent.click(getByTestId("test-btn"));
        expect(onClickMock).toHaveBeenCalledWith();
    });

    xit('should render a rectangle shaped button when type is passed as rect', () => {
        const { asFragment} = render(<Button testID="test-btn" type="rect"/>);
        expect(asFragment()).toMatchSnapshot();

    });
    xit('should render a circle shaped button when type is passed as round', () => {
        const { asFragment} = render(<Button testID="test-btn" type="round"/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('Should display the number of times the button is clicked along with the prop passed', () => {
        const { getByTestId} = render(<Button testID="test-btn" type="round" name = "123"/>);

        fireEvent.click(getByTestId("test-btn"));
        expect(getByTestId("test-btn")).toHaveTextContent("123 clicked 1 times");
    })
});