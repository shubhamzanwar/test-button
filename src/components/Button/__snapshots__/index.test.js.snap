// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should render correctly 1`] = `
<DocumentFragment>
  <button>
     clicked 0 times.
  </button>
</DocumentFragment>
`;

exports[`the button component should verify the type of button passed (rect) to the component 1`] = `undefined`;

exports[`the button component should verify the type of button passed (round) to the component 1`] = `undefined`;
